### PR TITLE
Add edit issues page

### PR DESCRIPTION
**Pull request by kahlouie**
Connects #6210 
This doesn't add the save button yet, it just adds the list of issues and preselects the issues that are already connected. Notably, it also doesn't edit the contentions on the ratings or nonratings end product. This'll only be for higher level reviews as I'm waiting on #6705

### CODE REVIEW

File: client/app/intake/pages/appeal/finish.jsx [hunk 1]
```diff
@@ -73,9 +73,8 @@ const NonRatedIssues = connect(
 )(NonRatedIssuesUnconnected);
 
 const RatedIssues = connect(
-  ({ appeal, intake }) => ({
-    intakeId: intake.id,
-    reviewState: appeal
+  ({ appeal }) => ({
```
Comment (leikkisa) [inline client/app/intake/pages/appeal/finish.jsx]: Nice refactor!

File: client/app/intake/util/index.js [hunk 2]
```diff
@@ -32,12 +32,18 @@ export const toggleIneligibleError = (hasInvalidOption, selectedOption) => (
...
-export const formatRatings = (ratings) => {
-  return _.keyBy(_.map(ratings, (rating) => {
+export const formatRatings = (ratings, requestIssues = []) => {
+  const result = _.keyBy(_.map(ratings, (rating) => {
     return _.assign(rating,
       { issues: _.keyBy(rating.issues, 'reference_id') }
     );
   }), 'profile_date');
+
+  _.forEach(requestIssues, (requestIssue) => {
```
Comment (leikkisa) [inline client/app/intake/util/index.js]: I like how you did this, should we also move this function into the "common" util file?

File: client/app/intakeCommon/components/NonRatedIssues.jsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import SearchableDropdown from '../../components/SearchableDropdown';
 import TextField from '../../components/TextField';
 import Button from '../../components/Button';
 import DateSelector from '../../components/DateSelector';
-import { ISSUE_CATEGORIES } from '../constants';
+import { ISSUE_CATEGORIES } from '../../intake/constants';
```
Comment (leikkisa) [inline client/app/intakeCommon/components/NonRatedIssues.jsx]: Let's move this into the common constants file after you re-base

File: client/app/intakeEdit/pages/selectIssues.jsx [hunk 4]
```diff
@@ -1,17 +1,47 @@
-import React, { Component } from 'react';
+import React from 'react';
 import { connect } from 'react-redux';
+import _ from 'lodash';
+import { bindActionCreators } from 'redux';
+import RatedIssuesUnconnected from '../../intakeCommon/components/RatedIssues';
+import { setIssueSelected } from '../../intake/actions/ama';
+import { FORM_TYPES } from '../../intake/constants';
```
Comment (leikkisa) [inline client/app/intakeEdit/pages/selectIssues.jsx]: This is going to affected by some recent changes on the landing page PR

File: client/app/intakeEdit/pages/selectIssues.jsx [hunk 5]
```diff
@@ -1,17 +1,47 @@
...
 
-class SelectIssues extends Component {
+// This page shouldn't matter that much which type of Review it is.
+class SelectIssues extends React.PureComponent {
   render() {
+    const {
+      veteranName,
```
Comment (leikkisa) [inline client/app/intakeEdit/pages/selectIssues.jsx]: Same here and other similar places

**Comment by kahlouie** [comment]
@leikkisa Changes were made.

**Pull request merged**